streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/api/Twitter.java

```diff
@@ -0,0 +1,521 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Implementation of all twitter interfaces using juneau.
+ */
+public class Twitter implements Followers, Friends, Statuses, Users {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Twitter.class);
+
+  private static Map<TwitterConfiguration, Twitter> INSTANCE_MAP = new ConcurrentHashMap<>();
+
+  private TwitterConfiguration configuration;
+
+  private ObjectMapper mapper;
+
+  private String rootUrl;
+
+  private CloseableHttpClient httpclient;
+
+  private HttpRequestInterceptor oauthInterceptor;
+
+  RestClient restClient;
+
+  private Twitter(TwitterConfiguration configuration) throws InstantiationException {
+    this.configuration = configuration;
+    this.rootUrl = TwitterProviderUtil.baseUrl(configuration);
+    this.oauthInterceptor = new TwitterOAuthRequestInterceptor(configuration.getOauth());
+    this.httpclient = HttpClientBuilder.create()
+        .addInterceptorFirst(oauthInterceptor)
+        .setDefaultRequestConfig(RequestConfig.custom()
+          .setConnectionRequestTimeout(5000)
+          .setConnectTimeout(5000)
+          .setSocketTimeout(5000)
+          .setCookieSpec("easy")
+          .build()
+        )
+        .setMaxConnPerRoute(20)
+        .setMaxConnTotal(100)
+        .build();
+
+//  TODO: juneau-6.2.0-incubating
+//  this.restClient = new RestClientBuilder()
+//        .httpClient(httpclient, true)
+//        .parser(JsonParser.class)
+//        .rootUrl(rootUrl)
+//        .retryable(
+//            configuration.getRetryMax().intValue(),
+//            configuration.getRetrySleepMs(),
+//            new TwitterRetryHandler())
+//        .build();
+    this.restClient = new RestClient()
+        .setHttpClient(httpclient)
+        .setParser(JsonParser.class)
+        .setRootUrl(rootUrl);
+
+    this.mapper = StreamsJacksonMapper.getInstance();
+  }
+
+  public static Twitter getInstance(TwitterConfiguration configuration) throws InstantiationException {
+    if (INSTANCE_MAP.containsKey(configuration) && INSTANCE_MAP.get(configuration) != null) {
+      return INSTANCE_MAP.get(configuration);
+    } else {
+      Twitter twitter = new Twitter(configuration);
+      INSTANCE_MAP.put(configuration, twitter);
+      return INSTANCE_MAP.get(configuration);
+    }
+  }
+
+  @Override
+  public List<Tweet> userTimeline(StatusesUserTimelineRequest parameters) {
+    try {
+//  TODO: juneau-6.2.0-incubating
+//      Statuses restStatuses = restClient.getRemoteableProxy("/statuses/user_timeline.json", Statuses.class);
+//      List<Tweet> result = restStatuses.userTimeline(parameters);
+//      return result;
+      URIBuilder uriBuilder = new URIBuilder()
+          .setPath("/statuses/user_timeline.json");
+      if( Objects.nonNull(parameters.getUserId()) && StringUtils.isNotBlank(parameters.getUserId().toString())) {
```

Comment:
I think the nonNull() checks are redundant because isNotBlank() "Checks if a String is not empty (""), not null and not whitespace only."